functional-test/src/test/groovy/org/rundeck/util/common/jobs/JobUtils.groovy

```diff
@@ -252,32 +253,46 @@ class JobUtils {
 
 
     /**
-     * Takes the job file and replaces all the words that start with 'xml-' using the provided args to upload it to the desired project.
+     * Takes the job template file and does textual replacement for the words that start with the 'xml-'  prefix.
+     * The default replacement arguments are used unless an overriden value is provided.
      *
-     * @param fileName The name of the file to import into test-files resources dir.
-     * @param args     Optional arguments as a Map. If not provided, default values will be used.
+     * @param fileName The name of the template file from the `test-files` resources dir.
+     * @param projectName project name to be used in the job file. Can be overriden in the argsOverrides.
+     * @param argsOverrides  argument keys-values that override the defaults.
      *                 Available keys:
-     *                 - "project-name": Name of the project (default: PROJECT_NAME)
+     *                 - "project-name": Name of the project (default: projectName parameter)
      *                 - "job-name": Name of the job (default: "job-test")
      *                 - "job-group-name": Name of the job group (default: "group-test")
      *                 - "job-description-name": Name of the job description (default: "description-test")
      *                 - "args": Arguments (default: "echo hello there")
      *                 - "2-args": Secondary arguments (default: "echo hello there 2")
      *                 - "uuid": UUID for the job (default: generated UUID)
+     *                 - "execution-enabled": Execution enabled flag (default: "true")
+     *                 - "schedule-enabled": Schedule enabled flag (default: "true")
+     *                 - "node-filter-include": Node filter include (default: ".*")
+     *                 - "node-filter-exclude": Node filter exclude (default: "")
+     *                 - "dispatch-rank-order": Dispatch rank order (default: "ascending")
      * @return The path of the updated temporary XML file.
      */
-    static def updateJobFileToImport = (String fileName, String projectName, Map args = [:]) -> {
-        if (args.isEmpty()) {
```

Comment:
This was unsustainable. Fortunately, the refactoring appears to be backward compatible.